### PR TITLE
service watchdog - add systemd watchdog for resiliency

### PR DESCRIPTION
**Pull request by bofh80**
I had 2 poller failures, both fired watchdog restarts, yet failed to poll and log errors.
systemctl restart librenms fixed both. I assume shutdown will lead to a service restart (restart=always is set on the service)

DO NOT DELETE THE UNDERLYING TEXT

#### Please note

> Please read this information carefully. You can run `./scripts/pre-commit.php` to check your code before submitting.

- [ ] Have you followed our [code guidelines?](http://docs.librenms.org/Developing/Code-Guidelines/)
- [ ] If my Pull Request does some changes/fixes/enhancements in the WebUI, I have inserted a screenshot of it.

#### Testers

If you would like to test this pull request then please run: `./scripts/github-apply <pr_id>`, i.e `./scripts/github-apply 5926`
After you are done testing, you can remove the changes with `./scripts/github-remove`.  If there are schema changes, you can ask on discord how to revert.

### CODE REVIEW

**Comment by mcineran** [comment]
My probes are restarting around once an hour with this patch. Which while seems odd, is better than randomly dying after 8. or 12 hours.  Not 100% sure why it would fire so often, testing welcome, but works for me.

**Comment by bofh80** [comment]
I sorted out problem with Redis and my strange service restarts have fixed themselves. This functioned as required and after a BARK, service watchdog restart, the logs still did not show any polling, then the systemd watchdog kicked in and restarted the service.
I should probably adjust the log messages to suggest if this has occurred there may be other underlying problems.

File: LibreNMS/service.py [hunk 1]
```diff
@@ -605,9 +605,12 @@ def logfile_watchdog(self):
...
             return
 
-        if logfile_mdiff > self.config.poller.frequency:
-            critical("BARK! Log file older than {}s, restarting service!".format(self.config.poller.frequency))
-            self.restart()
+        if self.config.watchdog_enabled:
+            if logfile_mdiff > self.config.poller.frequency:
+                critical("BARK! Log file older than {}s, restarting service!".format(self.config.poller.frequency))
+                self.restart()
+            else:
```
Comment (murrant) [inline LibreNMS/service.py:612]: The info message in the else does not line up with it does.  (It can only be reached if watchdog is enabled)

Comment (bofh80) [reply LibreNMS/service.py:612]: Does that look better?

**Comment by bofh80** [comment]
sorry this went wrong on the git rebase. i'll start a new branch.

**Pull request closed without merging**